app/main.py

```diff
@@ -1,25 +1,63 @@
+from __future__ import annotations
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: Any) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: SlideLimitationValidator, owner: Any) -> None:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: SlideLimitationValidator, value: Any) -> None:
+        if value in range(self.min_amount, self.max_amount + 1):
+            setattr(instance, self.protected_name, True)
+        else:
+            setattr(instance, self.protected_name, False)
```

Comment:
If value not in range you can raise an Exception

Reply:
done